cdisc_rules_engine/operations/library_model_column_order.py

```diff
@@ -0,0 +1,190 @@
+        Returns it as a Series of lists like:
+        0    ["STUDYID", "DOMAIN", ...]
+        1    ["STUDYID", "DOMAIN", ...]
+        2    ["STUDYID", "DOMAIN", ...]
+        ...
+
+        Length of Series is equal to the length of given dataframe.
+        The lists with column names are sorted
+        in accordance to "ordinal" key of library metadata.
+        """
+
+        # get variables metadata from the standard model
+        variables_metadata: List[
+            dict
+        ] = self._get_variables_metadata_from_standard_model()
+
+        # create a list of variable names in accordance to the "ordinal" key
+        variable_names_list = [
+            var["name"].replace("--", self.params.domain) for var in variables_metadata
+        ]
+        return variable_names_list
+
+    def _get_variables_metadata_from_standard_model(self) -> List[dict]:
+        """
+        Gets variables metadata for the given class and domain from cache.
+        The cache stores CDISC Library metadata.
+
+        Return example:
+        [
+            {
+               "label":"Study Identifier",
+               "name":"STUDYID",
+               "ordinal":"1",
+               "role":"Identifier",
+               ...
+            },
+            {
+               "label":"Domain Abbreviation",
+               "name":"DOMAIN",
+               "ordinal":"2",
+               "role":"Identifier"
+            },
+            ...
+        ]
+        """
+        # get model details from cache
+        cache_key: str = get_standard_details_cache_key(
+            self.params.standard, self.params.standard_version
+        )
+
+        standard_details: dict = self.cache.get(cache_key) or {}
+        model = standard_details.get("_links", {}).get("model")
+        model_type, model_version = self._get_model_type_and_version(model)
+        model_cache_key = get_model_details_cache_key(model_type, model_version)
+        model_details = self.cache.get(model_cache_key) or {}
+        domain_details = self._get_model_domain_metadata(
+            model_details, self.params.domain
+        )
+        class_title = domain_details["_links"]["parentClass"]["title"]
+        class_details = self._get_model_class_metadata(model_details, class_title)
+        variables_metadata = domain_details.get("datasetVariables", [])
+        variables_metadata.sort(key=lambda item: item["ordinal"])
+        class_name = class_details.get("name")
+        if class_name in DETECTABLE_CLASSES or class_name == FINDINGS_ABOUT:
+            # if the class is one of Interventions, Findings, Events, or Findings About
+            # -> get class variables instead of datasetVariables add
+            # General Observation class variables to variables metadata
+            variables_metadata = class_details.get("classVariables", [])
+            variables_metadata.sort(key=lambda item: item["ordinal"])
+
+            if class_name == FINDINGS_ABOUT:
```

Comment:
Is FINDINGS_ABOUT actually being detected anywhere? I don't see it in the BaseDataService.get_dataset_class.
If it needs to be added, we need to be careful that "Findings about" domains are still included when the scope specifies "findings"
Does this need to be a new ticket?

Reply:
In this operation the class is determined via library metadata. So if the domain appears in the findings about class it will be considered a findings about domain

Reply:
In the model, General Observations classes (Interventions, Events, Findings, and Findings About) don't have domains like they do in the IG.
Special Purpose, Trial Design, Study Reference, and Relationship classes have domains.

https://dev.cdisclibrary.org/browser/#/mdr/sdtm/2-0/classes/FindingsAbout

Reply:
@gcampion-cdisc if the model doesn't have the domain, then the class name would have to come from the ig correct? We can use the ig to get the class for the domain and then get the class variables from the model

Reply:
Updated to this logic:
1. Get domains parent class from IG
2. Get model class variables using the class found in the IG

if this logic is not sufficient then the engine will have to look at the actual data to determine the class. If that is the case we need an issue with clear criteria on how to determine if a dataset belongs to the FINDINGS_ABOUT class

Reply:
It shouldn't use the ig. If the domain name isn't found in one of the other model domains, it should use the topic variable the same way as the get_dataset_class method. That method should also be updated so that if a Findings class has an "--OBJ" variable, it should be considered "Findings About".

I would also update the `_contains_topic_variable` method to not use `ends_with`, but use 2-char wildcard matching instead. ABTRT is a topic variable, but ABCTRT is NOT a topic variable.

Reply:
Updated. Also update the _contains_topic_variable method to search the dataset for domain_name+topic_variable